Sources/Core/Error/DefaultError.swift

```diff
@@ -17,7 +17,7 @@ public struct DefaultError<UIErrorType: UIError>: AppError {
         intent: ErrorIntent = .indispensable
     ) {
         self.uiError = uiError
-        self.error = error
+        self.error = error as? NSError
```

Comment:
Hm, I'm not sure that this was correct. We may have to come back to this, but I can't see an obvious reason in SalesPro that this needed to be equatable.

Reply:
I recently encountered an obstacle on HomeStory where I had to jump through extra hoops to use `removeDuplicates()` because the repo result type was equatable. I would be in favor of enforcing that any errors that go through the repo are equatable and making the result type equatable. Another idea is we could create a Combine operator that converts a result type into an equatable result type by potentially wrapping the incoming error. Curious to hear your thoughts.